src/main/plugin/iso19139.ca.HNAP/schematron/schematron-rules-non-multilingual.sch

```diff
@@ -160,6 +160,19 @@
     />
   </xsl:function>
 
+  <!--Check email format-->
+  <xsl:function name="geonet:isEmailAddressFormat" as="xs:boolean">
+    <xsl:param name="inputEmailAdress"/>
+        <xsl:choose>
+            <xsl:when test="matches($inputEmailAdress, '^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9._%+-]+\.[a-zA-Z0-9]+$')">
```

Comment:
I'm not sure if that is the best email match? I have seen emails like `root@localhost` which would not work here - but also not a valid public email.

It you look at the following there is a full list of emails that are probably not supported.
https://github.com/asaskevich/govalidator/issues/131

And the following seems to be the regex used by html5 email check
https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address (It matches 12 emails from the list or 22 - while yours matches 5 from the list of 22)

Miles.O'Brian@example.com is a valid email that I have seen similar ones in our department.

I recommend that we follow html5 regex.  I would also include a comment indicating where the regex came from

Reply:
Sounds good. I did quick test with the supplied RegX in the html5, it works fine. I will update this function.

![image](https://user-images.githubusercontent.com/74916635/224367027-5decf96c-1b73-4577-81db-80d7280c5119.png)

Reply:
The pattern ^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$ is extremely hard to work into xslt. I have to use the Java Util function to implemented and it's working.